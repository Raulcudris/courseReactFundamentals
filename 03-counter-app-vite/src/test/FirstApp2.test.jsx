import { render, screen } from '@testing-library/react';
import { FirstApp } from '../FirstApp';

describe('Pruebas en <FirstApp 2 />', () => {    
  const title = 'Hola, Soy Raul';
  const subTitle = 'Soy un subtitulo';

  test('debe de hacer match con el snapshot', () => { 
        const {  container } = render(<FirstApp title={ title } />);
        expect( container ).toMatchSnapshot();
    });

    test('debe de mostar el mensaje "Hola, Soy Raul"', () => { 
      render(<FirstApp title={ title } />);
      expect( screen.getAllByText(title)).toBeTruthy();
      //screen.debug();
    });

    test('debe de mostrar el titulo en un h1', () => { 
      render(<FirstApp title={ title } />);
      expect( screen.getByRole('heading',{ level: 1}).innerHTML ).toContain( title );
    });

    test('debe de mostrar el subtitulo enviado por props', () => { 
       const { getAllByText } = render(
                    <FirstApp 
                    title={ title }
                    subTitle={subTitle} 
                    />);

       expect( getAllByText(subTitle).length ).toBe(1);
     });
    
 });