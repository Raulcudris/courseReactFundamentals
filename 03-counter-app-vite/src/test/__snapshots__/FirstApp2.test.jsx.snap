// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp 2 /> debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, Soy Raul
  </h1>
  <p>
    No hay subTitulo
  </p>
  <p>
    Ra√∫l Cudris
  </p>
</div>
`;
